### PR TITLE
qa: refactor health-ok.sh, drop health-{mds,rgw}.sh

### PR DESCRIPTION
**Pull request by smithfarm**
Now that we have the ceph.restart orchestration for testing the restart
functionality, we can simplify the integration testing scripts.

The actual triggering of the ceph.restart orchestration will be done
in the teuthology test yaml, which lives in the Ceph repo.

We retain `health-nfs-ganesha.sh` as a separate script for now, to 
keep `health-ok.sh` as simple as possible for as long as possible.

### CODE REVIEW

**Comment by smithfarm** [comment]
Hm, currently the scripts don't know how to deploy mds without a client (non-storage) node present. Fixing.

**Comment by smithfarm** [comment]
@jschmid1 Is the lint failure expected?

**Comment by smithfarm** [comment]
NOTE: setting DNM because this PR will break the teuthology `suse/smoke` suite, but otherwise I believe this PR is ready for review.

I'll prepare a downstream Ceph PR to update `suse/smoke` to work with this PR. Then we can merge both at once.

**Comment by smithfarm** [comment]
Tested so far:

On a single-node Salt cluster with 4 external disks, ran:

health-ok.sh --mds --rgw
salt-run state.orch ceph.restart

All green.

**Comment by smithfarm** [comment]
@jschmid1 As you can see, I'm removing the `ceph.restart` testing from these scripts. 

I have a Ceph branch: https://github.com/SUSE/tree/wip-qa-deepsea-func

I haven't opened a Ceph PR yet because that branch is WIP. But when it's ready it will have tests that reproduce the ceph-restart functional testing that is being removed from the health-ok, etc. scripts by this PR.

**Comment by jschmid1** [comment]
> @jschmid1 Is the lint failure expected?

I saw it failing initially for exactly what you are getting:

```
17:35:04 ************* Module srv.salt._modules.rgw
17:35:04 E: 22, 0: No name 's3' in module 'boto' (no-name-in-module)
17:35:04 E: 24, 0: No name 'exception' in module 'boto' (no-name-in-module)
```

those disappear on the next iteration ..

**Comment by jschmid1** [comment]
> I have a Ceph branch: https://github.com/SUSE/tree/wip-qa-deepsea-func

 I get a 404 on that one

**Comment by jschmid1** [comment]
> I'll prepare a downstream Ceph PR to update suse/smoke to work with this PR. Then we can merge both at once.

Let me know when to hit the merge button

**Comment by smithfarm** [comment]
@jschmid1 Sorry, fixed that comment so it reads correctly:

I have a Ceph branch: https://github.com/SUSE/ceph/tree/wip-qa-deepsea-func

File: qa/suites/basic/health-ok.sh [hunk 1]
```diff
@@ -81,22 +86,12 @@ ceph_log_grep_enoent_eaccess
...
-    change_osd_conf
-    change_mon_conf
-    # construct and spread config
-    run_stage_3 "$CLI"
-    restart_services
-    mon_restarted "0" # 0 means restarted
-    osd_restarted "0"
-    # make sure still in HEALTH_OK
-    ceph_cluster_status
-    ceph_health_test
```
Comment (smithfarm) [inline qa/suites/basic/health-ok.sh]: @jschmid1 Are you OK to drop this test of the restart functionality? Maybe the test should be re-implemented under `ceph.smoketests` before we really drop it?

**Comment by smithfarm** [comment]
Test **running** http://137.74.25.20:8081/ubuntu-2018-07-25_21:38:58-suse:smoke:functional-ses6---basic-openstack/35

**Comment by jschmid1** [comment]
> Test running http://137.74.25.20:8081/ubuntu-2018-07-25_21:38:58-suse:smoke:functional-ses6---basic-openstack/35

@smithfarm looks green to me :)

File: qa/suites/basic/health-ok.sh (hunk 1 above)
Comment (jschmid1) [reply qa/suites/basic/health-ok.sh]: Getting rid of it now means that we have at least for a short period of time no restart tests, right?
In that case I'd keep them in for now and remove them _after_ we implemented the restart.functests

**Comment by smithfarm** [comment]
@jschmid1 Yeah, sorry, I was originally under the impression that `ceph.smoketests` already includes the restart tests.

I will reinstate the restart tests (and "--mini") into health-ok.sh.

**Comment by jschmid1** [comment]
> @jschmid1 Yeah, sorry, I was originally under the impression that ceph.smoketests already includes the restart tests.

For master, we do. But do we already run those?

**Comment by smithfarm** [comment]
@jschmid1 So, to recapitulate, this PR:

1. drops health-mds.sh and health-rgw.sh
1. drops the restart tests that were in health-ok.sh
1. adds `--mds` and `--rgw` options to health-ok.sh, making it a more generic deployment script
1. makes the final re-run of stage 0 in health-nfs-ganesha.sh unconditional

In parallel, there will be a SUSE/ceph PR that will

1. add a test that uses health-ok.sh to deploy a cluster with rgw and mds roles, and runs `ceph.smoketests` on that cluster
2. revamp the mds and rgw CI tests to work with this PR

Sound right to you?

**Comment by jschmid1** [comment]
> @jschmid1 Would it be OK to rename ceph.smoketests (and everything under it) to ceph.functests.1node as part of this PR? That would clear the way for implementing ceph.functests.2nodes . . .

Yes, that'd be fine with me.

Currently everything is only required to have at least one node.
In the future the migration smoketests will require at least two

**Comment by smithfarm** [comment]
@jschmid1 OK, added smoketests rename commit - as a result, this PR now conflicts with #1216 so adding @swiftgist as reviewer.

**Comment by smithfarm** [comment]
@jschmid1 Latest test run http://137.74.25.20:8081/ubuntu-2018-07-26_22:31:13-suse:smoke-ses6---basic-openstack/

The only failures are in the tests that run `functests.1node.{migrate,replace}` which are not expected to succeed yet. I also examined the successful jobs and they seem to be doing what I expected.

So :+1: to merge from my side.

**Pull request merged**